Github Actons fails to install

### Note from maintainers

Make sure you are running latest Playwright version. Versions prior to 1.45.0 running on Ubuntu 24.04+ may cause the error described below.
___
All of a sudden a week or so ago, github CI has begun to fail installing playwright. even running old PRs that pass, now no longer build and run. Apologies if I am doing something wrong as I cannot find any other mention of these failures online. But I can't figure out what might be off. I can't be the only one!

yml
```
name: E2E tests

on:
  deployment_status:
jobs:
  run-e2es:
    if: github.event_name == 'deployment_status' && github.event.deployment_status.state == 'success' && github.event.deployment_status.environment != 'Production'
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: '20'
      - name: Install dependencies
        run: yarn install --frozen-lockfile
      - name: Install playwright
        run: npx playwright install --with-deps
      - name: Run tests
        run: npx playwright test
        env:
          NEXT_PUBLIC_BASE_URL: ${{ github.event.deployment_status.environment_url }}
```

and the error
```
Run npx playwright install --with-deps
  
Installing dependencies...
Switching to root user to install dependencies...
Get:1 file:/etc/apt/apt-mirrors.txt Mirrorlist [1[4](https://github.com/getrupt/dashboard/actions/runs/12796775812/job/35677191434?pr=212#step:5:4)2 B]
Hit:2 http://azure.archive.ubuntu.com/ubuntu noble InRelease
Hit:6 https://packages.microsoft.com/repos/azure-cli noble InRelease
Hit:7 https://packages.microsoft.com/ubuntu/2[4](https://github.com/getrupt/dashboard/actions/runs/12796775812/job/35677191434?pr=212#step:5:5).04/prod noble InRelease
Get:3 http://azure.archive.ubuntu.com/ubuntu noble-updates InRelease [126 kB]
Hit:4 http://azure.archive.ubuntu.com/ubuntu noble-backports InRelease
Hit:[5](https://github.com/getrupt/dashboard/actions/runs/12796775812/job/35677191434?pr=212#step:5:6) http://azure.archive.ubuntu.com/ubuntu noble-security InRelease
Fetched 12[6](https://github.com/getrupt/dashboard/actions/runs/12796775812/job/35677191434?pr=212#step:5:7) kB in 1s (193 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
Package libasound2 is a virtual package provided by:
  liboss4-salsa-asound2 4.2-build2020-1ubuntu3
  libasound2t64 1.2.11-1build2 (= 1.2.11-1build2)
Package libicu[7](https://github.com/getrupt/dashboard/actions/runs/12796775812/job/35677191434?pr=212#step:5:8)0 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
E: Package 'libasound2' has no installation candidate
E: Package 'libicu70' has no installation candidate
E: Unable to locate package libffi7
E: Unable to locate package libx264-163
Failed to install browsers
Error: Installation process exited with code: [10](https://github.com/getrupt/dashboard/actions/runs/12796775812/job/35677191434?pr=212#step:5:11)0
Error: Process completed with exit code 1.
```

@yury-s we're experiencing the same problem as @marcrupt. 

Here's our CI workflow file: https://github.com/NASA-AMMOS/aerie-ui/blob/develop/.github/workflows/test.yml#L65
The relevant section is:
```
runs-on: ubuntu-latest
    environment: ${{ inputs.env || 'test-workflow' }}
    steps:
      - name: Checkout Repo (UI)
        uses: actions/checkout@v4
      - name: Setup Node (UI)
        uses: actions/setup-node@v4
        with:
          node-version: 'lts/*'
          cache: npm
     ...
      - name: Install Playwright Dependencies (Test - e2e)
        run: npx playwright install chromium --with-deps
      - name: Test (e2e)
        run: npm run test:e2e
```

The output of `Set up Job` is:
```
Current runner version: '2.321.0'
Operating System
  Ubuntu
  [2](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:1:2)4.04.1
  LTS
Runner Image
  Image: ubuntu-24.04
  Version: 20250105.1.0
  Included Software: https://github.com/actions/runner-images/blob/ubuntu24/20250105.1/images/ubuntu/Ubuntu2404-Readme.md
  Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu24%2F20250105.1
Runner Image Provisioner
GITHUB_TOKEN Permissions
Secret source: Actions
Prepare workflow directory
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v4' (SHA:11bd71901bbe5b16[3](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:1:3)0ceea73d2759736[4](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:1:4)c9af683)
Download action repository 'actions/setup-node@v4' (SHA:39370e3970a6d0[5](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:1:5)0c480ffad4ff0ed4d3fdee5af)
Download action repository 'actions/upload-artifact@v4' (SHA:[6](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:1:7)5c4c4a1ddee5b[7](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:1:8)2f698fdd19549f0f0fb45cf0[8](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:1:9))
Complete job name: e2e-test
```

I've added `cat /etc/os-release` as a step before the playwright install. The output of that call was:
```
Run cat /etc/os-release
PRETTY_NAME="Ubuntu 24.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04.1 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo
```

And the failure on the playwright install step is:
```
Run npx playwright install chromium --with-deps
  
Installing dependencies...
Switching to root user to install dependencies...
Get:1 file:/etc/apt/apt-mirrors.txt Mirrorlist [142 B]
Hit:2 http://azure.archive.ubuntu.com/ubuntu noble InRelease
Hit:6 https://packages.microsoft.com/repos/azure-cli noble InRelease
Get:7 https://packages.microsoft.com/ubuntu/24.04/prod noble InRelease [3600 B]
Get:3 http://azure.archive.ubuntu.com/ubuntu noble-updates InRelease [126 kB]
Get:4 http://azure.archive.ubuntu.com/ubuntu noble-backports InRelease [126 kB]
Get:5 http://azure.archive.ubuntu.com/ubuntu noble-security InRelease [126 kB]
Get:8 https://packages.microsoft.com/ubuntu/24.04/prod noble/main arm64 Packages [11.6 kB]
Get:9 https://packages.microsoft.com/ubuntu/24.04/prod noble/main armhf Packages [6423 B]
Get:10 https://packages.microsoft.com/ubuntu/24.04/prod noble/main amd64 Packages [17.1 kB]
Get:11 http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 Packages [775 kB]
Get:12 http://azure.archive.ubuntu.com/ubuntu noble-updates/main Translation-en [176 kB]
Get:13 http://azure.archive.ubuntu.com/ubuntu noble-updates/main amd64 Components [151 kB]
Get:14 http://azure.archive.ubuntu.com/ubuntu noble-updates/universe amd64 Packages [974 kB]
Get:15 http://azure.archive.ubuntu.com/ubuntu noble-updates/universe amd64 Components [309 kB]
Get:16 http://azure.archive.ubuntu.com/ubuntu noble-updates/restricted amd64 Components [212 B]
Get:17 http://azure.archive.ubuntu.com/ubuntu noble-updates/multiverse amd64 Components [940 B]
Get:18 http://azure.archive.ubuntu.com/ubuntu noble-backports/main amd64 Components [[20](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:10:20)8 B]
Get:19 http://azure.archive.ubuntu.com/ubuntu noble-backports/universe amd64 Components [11.7 kB]
Get:[20](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:10:21) http://azure.archive.ubuntu.com/ubuntu noble-backports/restricted amd64 Components [[21](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:10:22)6 B]
Get:21 http://azure.archive.ubuntu.com/ubuntu noble-backports/multiverse amd64 Components [212 B]
Get:[22](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:10:23) http://azure.archive.ubuntu.com/ubuntu noble-security/main amd64 Packages [586 kB]
Get:[23](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:10:24) http://azure.archive.ubuntu.com/ubuntu noble-security/main Translation-en [114 kB]
Get:24 http://azure.archive.ubuntu.com/ubuntu noble-security/main amd64 Components [7[25](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:10:26)6 B]
Get:25 http://azure.archive.ubuntu.com/ubuntu noble-security/universe amd64 Packages [800 kB]
Get:[26](https://github.com/NASA-AMMOS/aerie-ui/actions/runs/12815536814/job/35734407936?pr=1595#step:10:27) http://azure.archive.ubuntu.com/ubuntu noble-security/universe amd64 Components [52.0 kB]
Get:27 http://azure.archive.ubuntu.com/ubuntu noble-security/restricted amd64 Components [212 B]
Get:28 http://azure.archive.ubuntu.com/ubuntu noble-security/multiverse amd64 Components [212 B]
Fetched 4376 kB in 1s (6304 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
Package libasound2 is a virtual package provided by:
  liboss4-salsa-asound2 4.2-build2020-1ubuntu3
  libasound2t64 1.2.11-1build2 (= 1.2.11-1build2)
E: Package 'libasound2' has no installation candidate
Failed to install browsers
Error: Installation process exited with code: 100
Error: Process completed with exit code 1.
```

--- a/packages/playwright-core/src/server/registry/dependencies.ts
+++ b/packages/playwright-core/src/server/registry/dependencies.ts
@@ -95,7 +95,7 @@ export async function installDependenciesLinux(targets: Set<DependencyGroup>, dr
   for (const target of targets) {
     const info = deps[platform];
     if (!info) {
-      console.warn(`Cannot install dependencies for ${platform}!`);  // eslint-disable-line no-console
+      console.warn(`Cannot install dependencies for ${platform} with Playwright ${getPlaywrightVersion()}!`);  // eslint-disable-line no-console
       return;
     }
     libraries.push(...info[target]);

--- a/packages/playwright-core/src/utils/hostPlatform.ts
+++ b/packages/playwright-core/src/utils/hostPlatform.ts
@@ -74,14 +74,18 @@ function calculatePlatform(): { hostPlatform: HostPlatform, isOfficiallySupporte
     // KDE Neon is ubuntu-based and has the same versions.
     // TUXEDO OS is ubuntu-based and has the same versions.
     if (distroInfo?.id === 'ubuntu' || distroInfo?.id === 'pop' || distroInfo?.id === 'neon' || distroInfo?.id === 'tuxedo') {
-      const isOfficiallySupportedPlatform = distroInfo?.id === 'ubuntu';
-      if (parseInt(distroInfo.version, 10) <= 19)
+      const isUbuntu = distroInfo?.id === 'ubuntu';
+      const version = distroInfo?.version;
+      const major = parseInt(distroInfo.version, 10);
+      if (major < 20)
         return { hostPlatform: ('ubuntu18.04' + archSuffix) as HostPlatform, isOfficiallySupportedPlatform: false };
-      if (parseInt(distroInfo.version, 10) <= 21)
-        return { hostPlatform: ('ubuntu20.04' + archSuffix) as HostPlatform, isOfficiallySupportedPlatform };
-      if (parseInt(distroInfo.version, 10) <= 22)
-        return { hostPlatform: ('ubuntu22.04' + archSuffix) as HostPlatform, isOfficiallySupportedPlatform };
-      return { hostPlatform: ('ubuntu24.04' + archSuffix) as HostPlatform, isOfficiallySupportedPlatform };
+      if (major < 22)
+        return { hostPlatform: ('ubuntu20.04' + archSuffix) as HostPlatform, isOfficiallySupportedPlatform: isUbuntu && version === '20.04' };
+      if (major < 24)
+        return { hostPlatform: ('ubuntu22.04' + archSuffix) as HostPlatform, isOfficiallySupportedPlatform: isUbuntu && version === '22.04' };
+      if (major < 26)
+        return { hostPlatform: ('ubuntu24.04' + archSuffix) as HostPlatform, isOfficiallySupportedPlatform: isUbuntu && version === '24.04' };
+      return { hostPlatform: ('ubuntu' + distroInfo.version + archSuffix) as HostPlatform, isOfficiallySupportedPlatform: false };
     }
     // Linux Mint is ubuntu-based but does not have the same versions
     if (distroInfo?.id === 'linuxmint') {

